### PR TITLE
Add OrcaContext and make spark default read file backend

### PR DESCRIPTION
**Pull request by hkvision**
- Move properties to OrcaContext.
@jason-dai Since log_output is used in init_nncontext, it can't be moved to OrcaContext?
- Make spark the default backend for reading csv/json files.
- When error occurs, add a message to hint users to switch backend.

### CODE REVIEW

File: pyzoo/zoo/orca/common.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#
+
+
+class OrcaContextMeta(type):
+
+    _orca_eager_mode = True
+    _orca_pandas_read_backend = "spark"
+
+    @property
+    def orca_eager_mode(cls):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/common.py:24]: change to `_eager_mode`, as this is supposed to be an internal property

File: pyzoo/zoo/orca/common.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        return cls._orca_eager_mode
+
+    @orca_eager_mode.setter
+    def orca_eager_mode(cls, value):
+        assert isinstance(value, bool), "orca_eager_mode should either be True or False"
+        cls._orca_eager_mode = value
+
+    @property
+    def orca_pandas_read_backend(cls):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/common.py:37]: change to `pandas_read_backend`

File: pyzoo/zoo/orca/common.py (hunk 1 above)
Comment (hkvision) [reply pyzoo/zoo/orca/common.py:24]: OK. So eager_mode is internal (generally not to be called by users) and pandas_read_backend is external?

File: pyzoo/zoo/common/nncontext.py [hunk 3]
```diff
@@ -115,8 +115,6 @@ def init_spark_on_yarn(hadoop_conf,
 class ZooContextMeta(type):
 
     _log_output = False
```
Comment (jason-dai) [inline pyzoo/zoo/common/nncontext.py:117]: we should either move `log_output` to `OrcaContext`, or have `OrcaContext.log_output` to call `ZooContext.log_output`

File: pyzoo/zoo/orca/common.py (hunk 1 above)
Comment (jason-dai) [reply pyzoo/zoo/orca/common.py:24]: yes

File: pyzoo/zoo/common/nncontext.py (hunk 3 above)
Comment (hkvision) [reply pyzoo/zoo/common/nncontext.py]: All review comments addressed, take another look? @jason-dai

File: pyzoo/zoo/common/nncontext.py [hunk 4]
```diff
@@ -112,30 +112,17 @@ def init_spark_on_yarn(hadoop_conf,
...
-        if SparkContext._active_spark_context is not None:
-            raise AttributeError("log_output cannot be set after SparkContext is created."
-                                 " Please set it before init_nncontext, init_spark_on_local"
-                                 "or init_spark_on_yarn")
-        cls._log_output = value
-
-
-class ZooContext(metaclass=ZooContextMeta):
-    pass
+    def log_output(self):
```
Comment (jason-dai) [inline pyzoo/zoo/common/nncontext.py:118]: if you have moved `log_output` to `OrcaContext`, then we can just eliminate `ZooContext`

Comment (hkvision) [reply pyzoo/zoo/common/nncontext.py:118]: Then is it strange to call OrcaContext outside Orca?

Comment (jason-dai) [reply pyzoo/zoo/common/nncontext.py:118]: I think there are two options:
1) Move everything to OrcaContext and remove ZooContext
2) Keep the property in ZooContext, and have OrcaContext call ZooContext

Comment (hkvision) [reply pyzoo/zoo/common/nncontext.py]: Updated. Take another look @jason-dai ?

**Comment by jason-dai** [comment]
LGTM

**Comment by hkvision** [comment]
jenkins: http://10.239.47.210:18888/job/ZOO-PR-Validation/3838/
One failure is due to node memory, not related to this PR.

**Pull request merged**